### PR TITLE
add proxy to package.json so that frontend can access the backend endpoints when running the frontend locally

### PR DESCRIPTION
**Pull request by mohitb3**
Using this `proxy` that's built into create-react-app, when running the frontend locally it can access the backend endpoints. https://create-react-app.dev/docs/proxying-api-requests-in-development/

If you pull the branch make sure you're on the booz vpn and restart the frontend. This way we don't have to use the mock files when running the frontend locally (at least for the endpoints which are already up and running).

### CODE REVIEW

**Comment by brian428** [comment]
Hmm I'm not sure we want to go this route. The `getRestUrl()` is meant to be sort of a "one-stop shop" for getting the URL a rest bundle should use. It will return one of 3 URLs, depending on how the app is running:

- In mock mode (`npm run mock`) it will return a URL to the mock JSON file
- In development mode (`npm start`), it will return a URL pointing at a local Docker container that is running the Go API (unless `mockOverrideFlag` is true, which forces it to return the mock JSON URL)
- In production mode, it will return the URL for the Go app on AWS (unless `mockOverrideFlag` is true, which forces it to return the mock JSON URL)

I also think there's some benefit to using `getRestUrl()` consistently, so we don't have some rest bundles using only a hardcoded URL, while others work differently by using `getRestUrl()`.

That said, I can also see the advantage of being able to hit the prod REST API from our local machines in certain cases (like testing the load mask).

What about adding another env file (.env.dev-aws) and associated npm script `npm run dev-aws`? .env.dev-aws would contain something like `REACT_APP_DEV_AWS_MODE=true`. Then, the `bundle-utils` file could be modified to include something like this:

```
export function isDevAwsMode( forceDevAwsMode ) {
  if( forceDevAwsMode === true ) return true;
  return process.env.REACT_APP_DEV_AWS_MODE === "true";
}

//....other existing code....

/**
 * Builds a REST URL. When not in mock mode, it uses the `liveUrl`. In mock mode, uses the `mockUrl`. If the
 * optional `mockOverrideFlag` is set to true, forces use of the `mockUrl`. If set to false, forces use of the `liveUrl`.
 * @param liveUrl
 * @param mockUrl
 * @param [mockOverrideFlag]
 * @returns {string}
 */
export function getRestUrl( liveUrl, mockUrl, mockOverrideFlag  ) {
  let useMockUrl = isMockMode();
  let useDevUrl = isDevelopmentMode();
  
  if( mockOverrideFlag === true || mockOverrideFlag === false ) useMockUrl = mockOverrideFlag;
  else if( isDevAwsMode() ) useDevUrl = useMockUrl = false;
  
  const baseUrl = useDevUrl
    ? useMockUrl ? `${ process.env.PUBLIC_URL }/mockdata` : `http://localhost:3030`
    : useMockUrl ? `${ process.env.PUBLIC_URL }/mockdata` : `https://api.rsgis.dev/development`;

  if( useMockUrl ) return `${ baseUrl }${ mockUrl }`;
  else return `${ baseUrl }${ liveUrl }`;
}
```

So basically, if we run the app with `rpm run dev-aws`, it will use the live AWS URL unless the `mockOverridFlag` is explicitly set in that call to `getRestUrl()`.

Seems like that might give us the best of both worlds, with the ability to run in mock mode, dev-aws mode, development mode (using local Go docker container), or production mode.

Thoughts?

**Comment by mohitb3** [comment]
No that's okay, we can leave it the way it is in develop. I think adding the `dev-aws` will over complicate things, it's probably simpler to just manually hit the dev api endpoint if necessary.
